src/write_hdf5_xml/write_hdf5_xml.c

```diff
@@ -2118,17 +2124,26 @@ int main(int argc, char **argv)
     double slant_first = meta->sar->slant_range_first_pixel;
     double slant_center = slant_first + ns*xs/2.0;
     double slant_last = slant_first + ns*xs;
-    strcpy(beam_mode, "FBS");
-    strcpy(original_file, params->input_HH_file);
-    if (params->input_HV_file) {
-      strcpy(beam_mode, "FBD");
-      sprintf(original_file, "%s,%s", params->input_HH_file, 
-        params->input_HV_file);
+    if (strmp_case(meta->general->sensor, "ALOS") == 0) {
+      strcpy(beam_mode, "FBS");
+      strcpy(original_file, params->input_HH_file);
+      if (params->input_HV_file) {
+        strcpy(beam_mode, "FBD");
+        sprintf(original_file, "%s,%s", params->input_HH_file, 
+          params->input_HV_file);
+      }
+      if (params->input_VH_file && params->input_VV_file) {
+        strcpy(beam_mode, "PLR");
+        sprintf(original_file, "%s,%s,%s,%s", params->input_HH_file, 
+          params->input_HV_file, params->input_VH_file, params->input_VV_file);
+      }
     }
-    if (params->input_VH_file && params->input_VV_file) {
-      strcpy(beam_mode, "PLR");
-      sprintf(original_file, "%s,%s,%s,%s", params->input_HH_file, 
-        params->input_HV_file, params->input_VH_file, params->input_VV_file);
+    else {
+      strcpy(beam_mode, "STD");
+      if (params->input_HH_file)
+        strcpy(original_file, params->input_HH_file);
+      else if (params->input_VV_file)
+        strcpy(original_file, params->input_VV_file);
```

Comment:
Maybe??????
if (params->input_HV_file || params->input_VH_file) {
  asfPrintError("Special beam mode definition needed for this sensor: %s", meta->general->sensor);
}